proteuslib/tests/test_chlorination.py

```diff
@@ -0,0 +1,823 @@
+        assert isinstance(model.fs.thermo_params.component('H_+'), Cation)
+        assert 'OH_-' in model.fs.thermo_params.component_list
+        assert hasattr(model.fs.thermo_params, 'OH_-')
+        assert isinstance(model.fs.thermo_params.component('OH_-'), Anion)
+
+        assert 'OCl_-' in model.fs.thermo_params.component_list
+        assert hasattr(model.fs.thermo_params, 'OCl_-')
+        assert isinstance(model.fs.thermo_params.component('OCl_-'), Anion)
+
+        assert 'NH4_+' in model.fs.thermo_params.component_list
+        assert hasattr(model.fs.thermo_params, 'NH4_+')
+        assert isinstance(model.fs.thermo_params.component('NH4_+'), Cation)
+
+        assert 'NH2Cl' in model.fs.thermo_params.component_list
+        assert hasattr(model.fs.thermo_params, 'NH2Cl')
+        assert isinstance(model.fs.thermo_params.component('NH2Cl'), Solute)
+
+        assert 'NHCl2' in model.fs.thermo_params.component_list
+        assert hasattr(model.fs.thermo_params, 'NHCl2')
+        assert isinstance(model.fs.thermo_params.component('NHCl2'), Solute)
+
+        assert 'NCl3' in model.fs.thermo_params.component_list
+        assert hasattr(model.fs.thermo_params, 'NCl3')
+        assert isinstance(model.fs.thermo_params.NCl3, Solute)
+
+        assert 'NH3' in model.fs.thermo_params.component_list
+        assert hasattr(model.fs.thermo_params, 'NH3')
+        assert isinstance(model.fs.thermo_params.NH3, Solute)
+
+        assert 'HOCl' in model.fs.thermo_params.component_list
+        assert hasattr(model.fs.thermo_params, 'HOCl')
+        assert isinstance(model.fs.thermo_params.HOCl, Solute)
+
+        assert hasattr(model.fs.thermo_params, 'phase_list')
+        assert len(model.fs.thermo_params.phase_list) == 1
+        assert hasattr(model.fs.thermo_params, 'Liq')
+        assert isinstance(model.fs.thermo_params.Liq, AqueousPhase)
+
+    @pytest.mark.unit
+    def test_units(self, chlorination_obj):
+        model = chlorination_obj
+        assert_units_consistent(model)
+
+    @pytest.mark.unit
+    def test_dof(self, chlorination_obj):
+        model = chlorination_obj
+        assert (degrees_of_freedom(model) == 0)
+
+    @pytest.mark.unit
+    def test_stats(self, chlorination_obj):
+        model = chlorination_obj
+        assert (number_variables(model) == 314)
+        assert (number_total_constraints(model) == 87)
+        assert (number_unused_variables(model) == 82)
+
+    @pytest.mark.unit
+    def test_custom_log_power_law_eps_options(self, chlorination_obj):
+        model = chlorination_obj
+
+        assert hasattr(model.fs.thermo_params, 'reaction_H2O_Kw')
+        assert hasattr(model.fs.thermo_params.reaction_H2O_Kw, 'eps')
+
+        assert hasattr(model.fs.rxn_params, 'reaction_NH2Cl_K')
+        assert hasattr(model.fs.rxn_params.reaction_NH2Cl_K, 'eps')
+
+        eps = 1e-30
+
+        # Inherent reactions have eps in the 'thermo_params'
+        model.fs.thermo_params.reaction_H2O_Kw.eps.value = eps

```

Comment:
I am not sure what you are testing for here; you set the value for `eps` and then check that it has the value you just assigned.

Reply:
Is there a function I can call to set this value? Should I just not have any asserts after setting the value?

The model is very sensitive to eps and this argument needs to be set for the solve.

Reply:
Ah, OK. Setting the value is fine, but do you really need to have the following `asserts`? I thought you were trying to test something specific.